### PR TITLE
Closes #6037, fixing FreeBSD compilation

### LINKED ISSUE #6037: FreeBSD: numba/np/ufunc/workqueue.c:30:10: fatal error: 'alloca.h' file not found
So after fixing the FreeBSD build in llvmlite, I get a somewhat weird message upon trying to install numba.

The compilation error is:

```
numba/np/ufunc/workqueue.c:30:10: fatal error: 'alloca.h' file not found
```

Upon searching for alloca.h, I found a similar file in `/usr/local/include/glib-2.0/glib/galloca.h`.

Any suggestions?

### PR DESCRIPTION
**Pull request by karolyi**
See https://github.com/numba/numba/issues/6037

### CODE REVIEW

**Comment by karolyi** [comment]
@esc hey, here's another small change, for FreeBSD compatibility. This time, without build errors. I'll have to use my fork again until this gets merged.

**Comment by esc** [comment]
@karolyi thank you very much, I have added it to the queue for review.

**Review by stuartarchibald — APPROVED** [review]
Thanks for the patch. Tested on FreeBSD 12.1 with vendor provided LLVM 10, this patch fixes the build and on testing some subset of the Numba test suite it appears to be working at runtime.

File: numba/np/ufunc/workqueue.c [hunk 1]
```diff
@@ -27,7 +27,13 @@ race conditions.
 /* PThread */
 #include <pthread.h>
 #include <unistd.h>
+
+#if defined(__FreeBSD__)
```
Comment (stuartarchibald) [inline numba/np/ufunc/workqueue.c:31]: NOTE: for future, I think all BSDs are likely going to require this.

**Comment by stuartarchibald** [comment]
@karolyi Congratulations on your first contribution to Numba!

**Pull request merged**